apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/boot/AgentPackagePath.java

```diff
@@ -50,6 +50,9 @@ private static File findPath() throws AgentPackageNotFoundException {
         String classResourcePath = AgentPackagePath.class.getName().replaceAll("\\.", "/") + ".class";
 
         URL resource = ClassLoader.getSystemClassLoader().getResource(classResourcePath);
+        if (resource == null) {
+            resource = AgentPackagePath.class.getClassLoader().getResource(classResourcePath);
+        }
```

Comment:
```suggestion
        URL resource = AgentPackagePath.class.getClassLoader().getResource(classResourcePath);
```

I am wondering, is this enough? Why do we need a fail-back? I think in most cases, `AgentPackagePath` should be loaded by the expected classloader(appclassloader or moduleclassloader)

Reply:
I think it should be compatible. Please change and remove all your test codes. I could rerun the CI to verify this change.